Sources/Workspace/Workspace.swift

```diff
@@ -1029,51 +1030,52 @@ extension Workspace {
+            let inputNodes: [GraphLoadingNode] = self.root.manifests.map { manifest in
+                let identity = PackageIdentity(manifest.url)
+                let package = PackageReference(identity: identity.description, path: manifest.url, kind: manifest.packageKind)
                 inputIdentities.insert(package)
                 let node = GraphLoadingNode(manifest: manifest, productFilter: .everything)
                 return node
-            }) + root.dependencies.compactMap({ dependency in
+            } + self.root.dependencies.compactMap { dependency in
                 let url = workspace.config.mirrors.effectiveURL(forURL: dependency.url)
-                let identity = PackageReference.computeIdentity(packageURL: url)
-                let package = PackageReference(identity: identity, path: url)
+                let identity = PackageIdentity(url)
+                let package = PackageReference(identity: identity.description, path: url)
                 inputIdentities.insert(package)
                 guard let manifest = manifestsMap[identity] else { return nil }
                 let node = GraphLoadingNode(manifest: manifest, productFilter: dependency.productFilter)
                 return node
-            })
+            }
 
             var requiredIdentities: Set<PackageReference> = []
             _ = transitiveClosure(inputNodes) { node in
-                return node.manifest.dependenciesRequired(for: node.productFilter).compactMap({ dependency in
+                node.manifest.dependenciesRequired(for: node.productFilter).compactMap { dependency in
                     let url = workspace.config.mirrors.effectiveURL(forURL: dependency.url)
-                    let identity = PackageReference.computeIdentity(packageURL: url)
-                    let package = PackageReference(identity: identity, path: url)
+                    let identity = PackageIdentity(url)
+                    let package = PackageReference(identity: identity.description, path: url)
                     requiredIdentities.insert(package)
                     guard let manifest = manifestsMap[identity] else { return nil }
                     return GraphLoadingNode(manifest: manifest, productFilter: dependency.productFilter)
-                })
+                }
             }
             // FIXME: This should be an ordered set.
             requiredIdentities = inputIdentities.union(requiredIdentities)
 
-            let availableIdentities: Set<PackageReference> = Set(manifestsMap.map({
+            let availableIdentities: Set<PackageReference> = Set(manifestsMap.map {
                 let url = workspace.config.mirrors.effectiveURL(forURL: $0.1.url)
-                return PackageReference(identity: $0.key, path: url, kind: $0.1.packageKind)
-            }))
+                return PackageReference(identity: $0.key.description, path: url, kind: $0.1.packageKind)
```

Comment:
can we use the actual identity instead of its description?

Reply:
See https://github.com/apple/swift-package-manager/pull/3032#discussion_r520751255

Reply:
See https://github.com/apple/swift-package-manager/pull/3032#discussion_r521326243